File: websocket.js
const socket = new WebSocket('ws://localhost:8765');

// Connection opened event handler
socket.addEventListener('open', function (event) {
    console.log('WebSocket connected');
    
    // Send a message to the server
    socket.send('Hello Server!');
});

// Message from server event handler
socket.addEventListener('message', function (event) {
    console.log('Message from server:', event.data);
});

// Error event handler
socket.addEventListener('error', function (event) {
    console.error('WebSocket error:', event);
});

// Close event handler
socket.addEventListener('close', function (event) {
    console.log('WebSocket closed');
});

File: websocket.py
import asyncio
import websockets

# Define a WebSocket handler function
async def websocket_handler(websocket, path):
    while True:
        try:
            # Receive message from the client
            message = await websocket.recv()
            print(f"Received message from client: {message}")

            # Send message back to the client
            await websocket.send(f"Server received: {message}")

        except websockets.exceptions.ConnectionClosed:
            print("Client disconnected")
            break

# Start the WebSocket server
start_server = websockets.serve(websocket_handler, 'localhost', 8765)

# Run the server forever
asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()
